src/rovr/app.py

```
@@ -196,6 +197,42 @@ def on_mount(self) -> None:
             return
         self.theme = config["theme"]["default"]
         self.ansi_color = config["theme"]["transparent"]
+        # Load persisted UI state (best-effort)
+        try:
+            ui_state = load_ui_state()
+            # Apply show_hidden_files if saved (fallback to config otherwise)
+            if ui_state.get("show_hidden_files") is not None:
+                config["settings"]["show_hidden_files"] = bool(
+                    ui_state["show_hidden_files"]
+                )
+            # Apply footer visibility if saved
+            if ui_state.get("footer_visible") is not None:
+                if not ui_state["footer_visible"]:
+                    # start hidden
+                    self.query_one("#footer").add_class("hide")
+                else:
+                    self.query_one("#footer").remove_class("hide")
+            # Apply pinned sidebar visibility
+            if ui_state.get("pinned_sidebar_visible") is not None:
+                if not ui_state["pinned_sidebar_visible"]:
+                    self.query_one("#pinned_sidebar_container").add_class("hide")
+                else:
+                    self.query_one("#pinned_sidebar_container").remove_class("hide")
+            # Apply preview visibility
+            if ui_state.get("preview_visible") is not None:
+                if not ui_state["preview_visible"]:
+                    self.query_one("PreviewContainer").add_class("hide")
+                else:
+                    self.query_one("PreviewContainer").remove_class("hide")
+            # Compact mode
+            if ui_state.get("compact_mode") is not None:
+                if ui_state["compact_mode"]:
+                    self.add_class("compact")
+                else:
+                    self.remove_class("compact")
+        except Exception:
+            # best-effort: do not prevent startup on any failure
+            pass
```

please catch something, and display it to the user, so that it wont silently do nothing